xds/src/main/java/io/grpc/xds/LoadReportClient.java

```diff
@@ -129,6 +129,13 @@ void stopLoadReporting() {
     // Do not shutdown channel as it is not owned by LrsClient.
   }
 
+  /**
+   * Returns {@code true} if the client has already been started.
+   */
+  boolean isStarted() {
+    return started;
```

Comment:
Yeah, that should really check that it is in a sync context. I paid a bit less attention to it because it was just for the test, but I removed the method now, since it is wrong.

I do need to still add a way to determine if the LRS connection has been started. I made the lrsStream field package private and marked it as @VisibleForTesting. This doesn't guarantee thread safety either, but it works for the test, and this more surgical approach hopefully makes it clearer what this is just for.

Reply:
Ok I think this approach to testing is inherently flaky - the test might be faster than `XdsClientImpl` and make the assertion before the LRS client has been started. Ideally I would have access to the `SynchronizationContext` `XdsClientImpl` uses, but it is not exposed or allowed to be passed in. I might make the context accessible for testing or think of another way to synchronize.